shapes/typeIndex.shacl

```diff
@@ -0,0 +1,25 @@
+PREFIX sh: <http://www.w3.org/ns/shacl#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX solid: <http://www.w3.org/ns/solid/terms#>
+PREFIX : <#>
+
+:TypeIndexShape a sh:NodeShape ;
+    sh:targetClass solid:ListedDocument, solid:UnlistedDocument, solid:TypeIndex .
```

Comment:
TODO: introduce example with `solid:UnlistedDocument` in the spec as well. Should I create an issue for this?

Reply:
Yes, please. Thank you.